services/src/main/java/org/keycloak/protocol/oidc/grants/AuthorizationCodeGrantType.java

```diff
@@ -188,12 +192,18 @@ public Response process(Context context) {
             throw new CorsErrorResponseException(cors, OAuthErrorException.INVALID_GRANT, cpe.getErrorDetail(), Response.Status.BAD_REQUEST);
         }
 
+        String scopeParam = codeData.getScope();
+        ClientSessionContext clientSessionCtx = DefaultClientSessionContext.fromClientSessionAndScopeParameter(clientSession, scopeParam, session);
+
+        // OID4VCI: Process authorization_details using the processor
+        OID4VCAuthorizationDetailsProcessor oid4vciProcessor = new OID4VCAuthorizationDetailsProcessor(session, event, formParams, cors);
```

Comment:
Since this effectively introduces support to process RAR `authorization_details` I wonder if it makes sense to have a more generic implementation here. 

Perhaps we can have an interface like AuthorizationDetailsProcessor, with a DefaultAuthorizationDetailsProcessor implementation and a special OID4VCIAuthorizationDetailsProcessor implementation. 

Then we could then associate the AuthorizationDetailsProcessor implementation to the OIDCLoginProtocol and query it from there - if we detect that OpenID4VCI is used, we can re turn the OID4VCIAuthorizationDetailsProcessor.